contracts/delegatecall/AssetTransferToENS.sol

```diff
@@ -0,0 +1,92 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+pragma solidity ^0.8.20;
+
+import {ENS} from "@ensdomains/ens-contracts/contracts/registry/ENS.sol";
+import {AddrResolver} from "@ensdomains/ens-contracts/contracts/resolvers/profiles/AddrResolver.sol";
+
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {IERC721} from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import {IERC1155} from "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
+
+import {LibError} from "../library/LibError.sol";
+
+contract AssetTransferToENS {
+    using LibError for bytes;
+    using SafeERC20 for IERC20;
+
+    ENS immutable _ens;
+
+    constructor(ENS ens) {
+        _ens = ens;
+    }
+
+    function sendCallToENS(
+        bytes32 node,
+        uint256 value,
+        bytes memory payload
+    ) external {
+        address recipient = resolveENS(node);
+
+        bool success;
+        bytes memory returndata;
+        (success, returndata) = recipient.call{value: value}(payload);
+        if (!success) {
+            returndata.raise();
+        }
```

Comment:
We could create a library for `safecall`, although I think this code block is already easy to read on its own. Not sure if it improves readability by putting it into a library. Because when reading the codes, readers would need to jump to the library to see what the call is, and then jump back.

Reply:
changed my mind :)

Reply:
There's some degree of strictness to which we should follow the DRY mantra ("don't repeat yourself"). I think in this case, it helps the user abstract the complexity of low-level operations such as `.call` and `.delegatecall` and handling of return data as error data, which is not very native Solidity-like. Wrapping that in a Solidity library helps abstract implementation details. Also, there's DRY, because this code is repeated here and in `Application`.